### ISSUE TITLE
Feature request: allow specifying environment variables in YAML schema

### ISSUE THREAD

**Issue by nre-ableton** [wickedchicken/checklist#75]
I have some projects for which I would like run commands which require an associated Python virtualenv. I could run `checklist` from within the virtualenv, but I sometimes run git commands from a separate shell (or IDE) which isn't always aware of the IDE.

It turns out that assuming the virtualenv is already properly configured, you can just run the executables there directly. However, I sometimes run into path issues when some of the tests want to call executables from other pip modules.

Basically, what I'd like to do is be able to specify environment variables which would be passed to each automated test. I imagine this could look something like this:

```yaml
---
schema_version: 2
environment:
  VENV_BIN: "$HOME/caches/venv/my_project/bin"
  PATH: "$PATH:$VENV_BIN"
committing:
  automated:
    - python ./run.py check  # assumes $PATH is correct 
```

I'm not sure if the `PATH` definition above would work as expected unless the variables were expanded in the shell.

**Comment by wickedchicken** [comment]
Great idea! I'm cleaning out some of the cobwebs in the codebase right now and will think about how to best implement this.

**Comment by wickedchicken** [comment]
I researched this a bit, and thought I'd write some notes here:

* [duct_sh::sh_dangerous](https://docs.rs/duct_sh/0.1.0/duct_sh/fn.sh_dangerous.html) doesn't have an option for adding envars, which would be the ideal solution. [duct](https://docs.rs/duct/0.13.3/duct/) may have envar-handling capabilities, but I'm not sure if they can be chained up with `duct_sh` or have to be used within `duct`.
* Adding them into the command-string itself is error-prone (you'd have to determine the syntax for each shell) and won't work if commands include shell piping (unless you parse out the command and add the envars to every part of the pipe).
* [std::set_var](https://doc.rust-lang.org/std/env/fn.set_var.html) could work, but it looks like it's not threadsafe and would generally be unreliable.
* Shelling out to a separate process (and setting envars as part of that call) might work, but is annoying.

I'm going to investigate `duct` more to see if that can be used with `duct_sh`.

**Comment by wickedchicken** [comment]
I've requested that feature here: https://github.com/oconnor663/duct.rs/issues/87.
